### PR TITLE
Document Snaps data storage

### LINKED ISSUE #1262: Add Snaps storage feature page
Add topic under **Snaps > Features** describing how to use encrypted and unencrypted storage with `snap_manageState`. Link to the [manage state example package](https://github.com/MetaMask/snaps/tree/main/packages/examples/packages/manage-state). The storage cards on the Snaps intro page should be consolidated and link to this feature page.

### PR DESCRIPTION
**Pull request by alexandratran**
# Description

Add feature page on Snaps encrypted and unencrypted data storage.

## Issue(s) fixed

Fixes #1262

## Preview

https://docs.metamask.io/1262-snaps-storage/snaps/features/data-storage/

## Checklist

Complete this checklist before merging your PR:

- [x] If this PR contains a major change to the documentation content, I have added an entry to the top of the ["What's new?"](https://github.com/MetaMask/metamask-docs/blob/main/docs/whats-new.md) page.
- [ ] The proposed changes have been reviewed and approved by a member of the documentation team.

### CODE REVIEW

**Review by Montoya — APPROVED** [review]

File: snaps/features/data-storage.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+### 2. Use encrypted storage
+
+By default, [`snap_manageState`](../reference/snaps-api.md#snap_managestate) automatically encrypts
+data using a Snap-specific key before storing it on the user's disk, and automatically decrypts it
+when retrieved.
+This is useful to store sensitive information, such as passwords.
+
+The following example uses `snap_manageState` to store some data using the `update` operation, and
+retrieves the data at a later time using the `get` operation.
+Once there's no need to store the data anymore, it clears the Snap's state using the `clear` operation.
```
Comment (joaniefromtheblock) [inline snaps/features/data-storage.md:36]: I think refrain from using "once" as a synonym for after 
https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/o/once

```suggestion
When the data is no longer required, the Snap's state is cleared using the `clear` operation.
```

File: snaps/features/data-storage.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  params: { 
+    operation: "get",
+    encrypted: false,
+  },
+});
+
+console.log(persistedData);
+// { hello: "world" }
+
+// If there's no need to store data anymore, clear it out.
```
Comment (joaniefromtheblock) [inline snaps/features/data-storage.md:105]:
```suggestion
// If data storage is no longer necessary, clear it.
```

File: snaps/features/data-storage.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+// At a later time, get the stored data.
+const persistedData = await snap.request({
+  method: "snap_manageState",
+  params: { operation: "get" },
+});
+
+console.log(persistedData);
+// { hello: "world" }
+
+// If there's no need to store data anymore, clear it out.
```
Comment (joaniefromtheblock) [inline snaps/features/data-storage.md:57]:
```suggestion
// // If data storage is no longer necessary, clear it.
```

**Review by joaniefromtheblock — APPROVED** [review]
lgtm

**Review by subhamkumarr — APPROVED** [review]

**Pull request merged**